### PR TITLE
ENH: separate exposure calculation in perf attrib

### PR DESCRIPTION
**Pull request by vikram-narayan**
- add `compute_exposures` which computes exposures
- put warning logic and conversion of positions to long format in own functions

### CODE REVIEW

File: pyfolio/perf_attrib.py [hunk 1]
```diff
@@ -522,3 +491,149 @@ def plot_risk_exposures(exposures, ax=None,
     ax.set_title(title)
 
     return ax
+
+
+def _truncate_and_warn(returns,
```
Comment (yankees714) [inline pyfolio/perf_attrib.py]: Something like 'align' might be clearer here then 'truncate', to me truncate suggests something more like trimming the ends

File: pyfolio/perf_attrib.py [hunk 2]
```diff
@@ -120,133 +120,102 @@ def perf_attrib(returns,
...
+    # above, since get_turnover() expects positions in dollars.
+    positions = _stack_positions(positions, pos_in_dollars=pos_in_dollars)
 
-        else:
-            missing_stocks_displayed = (
-                "The following assets were missing factor loadings: {}."
-            ).format(list(missing_stocks))
-            avg_allocation_msg = "missing assets"
+    risk_exposures_portfolio = compute_exposures(positions, factor_loadings,
+                                                 stack_positions=False)
```
Comment (yankees714) [inline pyfolio/perf_attrib.py]: Can you put each of these on their own line?

File: pyfolio/perf_attrib.py (hunk 1 above)
Comment (vikram-narayan) [reply pyfolio/perf_attrib.py]: I agree, renamed to this

File: pyfolio/perf_attrib.py (hunk 2 above)
Comment (vikram-narayan) [reply pyfolio/perf_attrib.py]: put each on their own line

**Comment by vikram-narayan** [comment]
@yankees714 specified `ImportError`for a couple unrelated things - the upgraded flake8 version on travis fails for them

**Pull request merged**